runtime/sema/path_type.go

```diff
@@ -30,10 +30,6 @@ var PathType = &SimpleType{
 	Comparable:    false,
 	Exportable:    true,
 	Importable:    true,
-	IsSuperTypeOf: func(subType Type) bool {
```

Comment:
I think after this change, the `SimpleType.IsSuperTypeOf` functionality is no longer needed. The central implementation of the subtyping rules in `checkSubTypeWithoutEquality` seems better, so maybe we can just remove the function now. Wdyt? also cc @onflow/cadence

Reply:
+1 for removing it while at it, if it's not used anywhere else. Otherwise, it could be a separate PR for the refactor.

Reply:
The only other case is `StorableType`, but that should be easy to move to `checkSubTypeWithoutEquality` as well

Reply:
Remove in 8a16ed2b3.